### ISSUE TITLE
Problem building SVG with JS for browser use

### ISSUE THREAD

**Issue by saviomd** [FortAwesome/Font-Awesome#11982]
I'm trying to build a custom version with only the icons I'll actually use but I get this error on the dev tools console:
```
Uncaught ReferenceError: module is not defined
```

What I'm doing is concatenating the files I need in my js file (together with the rest of my code).
The files are:
```
node_modules/@fortawesome/fontawesome/index.js
node_modules/@fortawesome/fontawesome-free-solid/faFutbol.js
node_modules/@fortawesome/fontawesome-free-solid/faSquare.js
and so on
```
But it breaks here:
```
module.exports = { prefix: 'fas', iconName: 'futbol'
```

From the docs, I could see that these modules are made for node.js use (and I am aware that the browsers don't support modules).
Is it possible to make the SVG with JS modules published at npm compatible with the browser too?
Or I have to do something on my side to make it work? (I just concat everything in one file and uglify it, no babel, browserify or stuff like that)

**Comment by spointecker** [comment]
Hi,

have you found a solution to just load the icons you need in your project?

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community.

Does this help?

https://fontawesome.com/how-to-use/use-with-node-js#tree-shaking

**Comment by spointecker** [comment]
Hi @tagliala 
thanks for you quick response.
Unfortunately not really, cause the examples are not fully understandable.

We use grunt for asset packaging. So the main question is: is it possible to use grunt and not webpack? or, what do I need to get it working. :)

**Comment by tagliala** [comment]
Hi, unfortunately I don't know how to do that.

By the way, I'm not sure that your issue is the same of @saviomd's one. The first one doesn't look related to subsetting/tree-shaking

**Comment by saviomd** [comment]
@tagliala, it kind of is.
I was trying to build a subset of the library, using file concatenation via gulp.
I'm starting to explore es6, webpack, rollup and stuff like that and will try your solution using es6 import.
(but umd compatible files instead of commonjs would be very welcome :) )

**Comment by spointecker** [comment]
yes, that was my intention, to find a solution for subsetting icons without webpack etc.

**Comment by wgmv** [comment]
Also looking for subsetting - I'd like to avoid adding 400kb of icons I don't use.

**Comment by tagliala** [comment]
Subsetting: https://fontawesome.com/how-to-use/use-with-node-js#tree-shaking

**Comment by sensibleworld** [comment]
Doing a little management in Github and (batch) closing out potentially stale issues. If the above still needs attention, please let us know and we’ll re-open!
